x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/edit_policy.container.tsx

```diff
@@ -86,12 +90,19 @@ export const EditPolicy: React.FunctionComponent<Props & RouteComponentProps<Rou
     );
   }
 
+  const existingPolicy = getPolicyByName(policies, policyName);
+
   return (
-    <PresentationComponent
-      policies={policies}
-      history={history}
-      getUrlForApp={getUrlForApp}
-      policyName={policyName}
-    />
+    <EditPolicyContextProvider
+      value={{
+        isNewPolicy: !existingPolicy?.policy,
+        policyName,
+        policy: existingPolicy?.policy ?? defaultPolicy,
+        existingPolicies: policies,
+        getUrlForApp,
+      }}
+    >
+      <PresentationComponent history={history} />
```

Comment:
I would move `history` prop to context for consistency or alternatively, pass `isNewPolicy`, `existingPolicies` and `policyName` as props too. I think they are only used in `edit_policy.tsx`.

Reply:
I see your point, but would like to hold off on this change because it seems cleaner to me to have the `EditPolicyContextProvider` contain the values for the policy business logic. Most of those values are relevant to the "name" field of the policy and I would like to preserve the flexibility of being able to move that field into its own component if needed.